### ISSUE TITLE
Uint8Array 'invalid arguments' error when using array buffer from XMLHTTPRequest

### ISSUE THREAD

**Issue by samuknet** [servo/servo#12503]
Encountered some unusual results when using an XMLHTTPRequest with `responseType` set to `arraybuffer` - Servo throws an 'invalid arguments' error when trying to create a `Uint8Array` from the XMLHTTPRequests' array buffer response.  

Code:

```
<script>
var oReq = new XMLHttpRequest();
oReq.open('GET', 'https://www.wikipedia.org/portal/wikipedia.org/assets/img/Wikipedia-logo-v2.png', true);
oReq.responseType = 'arraybuffer';

oReq.onload = function (oEvent) {
    var aBuffer = oReq.response; // This response appears to be valid image data
    var byteArray = new Uint8Array(aBuffer); // This line causes 'invalid arguments' error to occur
};

oReq.send(null);    
</script>
```

Live example of the issue at:
[https://samuknet.github.io/test_cases/xhrUint8Array/](https://samuknet.github.io/test_cases/xhrUint8Array/)

Also was wondering in general what is the state of the implementation of Uint8Array/DataView/ArrayBuffer's in Servo?

**Comment by jdm** [comment]
Uint8/DataView/ArrayBuffer should all work as expected in JS. However, our support for APIs that accept these types as arguments is _very_ spotty. We should definitely look into cases where we return them and they don't behave correctly, however.

**Comment by Manishearth** [comment]
We don't support `oReq.responseType = 'arraybuffer';` yet, the response is a string.

This is blocked on getting proper arraybuffer bindings I think.

**Comment by nox** [comment]
We have bindings for those now.

**Comment by jdm** [comment]
This was fixed in #19352.
